test/compatibility/test_fast_truncate_disk_space.py

```diff
@@ -0,0 +1,175 @@
+        conn.close()
+
+    def write_fast_truncate_info(self):
+        import wiredtiger
+        from wiredtiger import stat
+
+        # Open the connection on the database directory containing the database
+        # generated by the older binary.
+        new_conn = wiredtiger.wiredtiger_open('.', 'create=false,statistics=(all)')
+
+        print(f'Running on {wiredtiger.wiredtiger_version()[0]}')
+
+        new_conn.set_timestamp('oldest_timestamp='+ self.timestamp_str(200))
+        new_conn.set_timestamp('stable_timestamp='+ self.timestamp_str(200))
+
+        session = new_conn.open_session()
+        session.begin_transaction()
+        
+        # Truncate a big range of rows.
+        start = session.open_cursor(self.uri, None)
+        start.set_key(10)
+        end = session.open_cursor(self.uri, None)
+        end.set_key(self.nentries - 10)
+        session.truncate(None, start, end, None)
+        start.close()
+        end.close()
+
+        session.commit_transaction('commit_timestamp=' + self.timestamp_str(250))
+
+        # Ensure that fast-truncate happened.
+        stat_cursor = session.open_cursor('statistics:', None, None)
+        fastdelete_pages = stat_cursor[stat.conn.rec_page_delete_fast][2]
+        self.assertGreater(fastdelete_pages, 0)
+
+        # Move the stable timestamp ahead of the fast-truncate timestamp so that
+        # recovery will not remove this fast-truncate information.
+        new_conn.set_timestamp('stable_timestamp='+ self.timestamp_str(300))
+
+        # Checkpoint, forcing address-deleted cells to be written.
+        session.checkpoint()
+
+        session.close()
+        new_conn.close()
+
+        # Re-open the connection and verify the object.
+        new_conn = wiredtiger.wiredtiger_open('.', 'create=false')
+        session = new_conn.open_session()
+        session.verify(self.uri)
+
+        session.close()
+        new_conn.close()
+
+    def read_fast_truncate_info_with_cleanup(self):
+        import wiredtiger
+
+        # Open the connection on the database directory containing the database
+        # generated by the newer binary which contains fast-truncate information.
+        new_conn = wiredtiger.wiredtiger_open('.', 'create=false')
+
+        print(f'Running on {wiredtiger.wiredtiger_version()[0]}')
+
+        disk_space_original = os.path.getsize("test_fast_truncate")
+
+        # Update some records in the truncate range to dirty the tree.
+        # Additionally, do multiple checkpoints.
+        session = new_conn.open_session()
+        c = session.open_cursor(self.uri)
+        c.set_key(50)
+        c.set_value("cleanup")
+        c.update()
+        c.reset()
+        session.checkpoint()
+
+        c.set_key(5000)
+        c.set_value("cleanup")
+        c.update()
+        c.reset()
+        session.checkpoint()
+
+        # Check that we are using less space than before after the pages
+        # containing fast-truncate information were cleaned up.
```

Comment:
Discussed offline, this can be tracked with the cc_pages_removed statistic but we'll need to raise another ticket first to fix the logic where it is incremented.